<?xml version="1.0"?>
<robot name="leadsense_camera">

    <link name="leadsense_left_camera">
        <visual>
            <origin xyz="0 0 0" rpy="0 1.57 0"/>
            <geometry>
                <cylinder radius="0.007" length="0.004"/>
            </geometry>
            <material name="dark_grey">
                <color rgba="0.1 0.1 0.1 1"/>
            </material>
        </visual>
    </link>
    <link name="leadsense_center">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://leadsense_ros/urdf/leadsenseL.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="grey">
                <color rgba="0.8 0.8 0.8 1"/>
            </material>
        </visual>
    </link>
    <link name="leadsense_right_camera">
        <visual>
            <origin xyz="0 0 0" rpy="0 1.57 0"/>
            <geometry>
                <cylinder radius="0.007" length="0.004"/>
            </geometry>
            <material name="dark_grey">
                <color rgba="0.1 0.1 0.1 1"/>
            </material>
        </visual>
    </link>

    <joint name="leadsense_center_joint" type="fixed">
        <parent link="leadsense_left_camera"/>
        <child link="leadsense_center"/>
        <origin xyz="0 0 0" rpy="-1.57 0 -1.57" />
    </joint>
    <joint name="leadsense_right_camera_joint" type="fixed">
        <parent link="leadsense_left_camera"/>
        <child link="leadsense_right_camera"/>
        <origin xyz="0 -0.12 0" rpy="0 0 0" />
    </joint>

</robot>
